lib/puppet/ssl/host.rb

```diff
@@ -227,7 +227,29 @@ def validate_certificate_with_key
   # Generate all necessary parts of our ssl host.
   def generate
     generate_key unless key
-    generate_certificate_request unless certificate_request
+    # ask indirector to find any existing requests and download them
+    existing_request = certificate_request
+
+    # if CSR downloaded from master, but the local keypair was just generated and
```

Comment:
I don't believe that agents ever fetch the CSR from the master, have you seen otherwise?